helmfile.d/charts/gatekeeper/templates/policies/tests/restrict-pod-disruption-budgets.rego

```diff
@@ -0,0 +1,742 @@
+        with input as pod_controllers[_]
+        with data.inventory as {"namespace": {namespace: {"policy/v1": {"PodDisruptionBudget": [generate_pdb(
+            "minAvailable",
+            5,
+            namespace,
+            selector
+        )]}}}}
+    ]
+    all(res)
+}
+
+test_bad_pod_controller_minAvailable_percent {
+    namespace := "default"
+    selector := {"matchLabels": {"key": "val"}}
+    labels := {"key": "val"}
+    pod_controllers := [pod_controller | pod_controller = input_wrap(generate_pod_controller(
+        pod_controller_groups_kinds[_].kind,
+        3,
+        namespace,
+        labels
+    ))]
+
+    res = [test | test = count(k8srestrictpoddisruptionbudgets.violation) == 1
+        with input as pod_controllers[_]
+        with data.inventory as {"namespace": {namespace: {"policy/v1": {"PodDisruptionBudget": [generate_pdb(
+            "minAvailable",
+            "90%",
+            namespace,
+            selector
+        )]}}}}
+    ]
+    all(res)
+}
+
+# --- Test replicaset under deployment as review object ---
+
+test_bad_replicaset_under_deployment_maxUnavailable_int {
+    namespace := "default"
+    selector := {"matchLabels": {"key": "val"}}
+    labels := {"key": "val"}
+
+    # The replicaset and PDB does not allow for disruption, but this should not violate since replicaset is under control of a deployment
+    res = [test | test = count(k8srestrictpoddisruptionbudgets.violation) == 0
+        with input as input_wrap(generate_replicaset_under_deployment(
+            3,
+            namespace,
+            labels
+        ))
+        with data.inventory as {"namespace": {namespace: {"policy/v1": {"PodDisruptionBudget": [generate_pdb(
+            "maxUnavailable",
+            0,
+            namespace,
+            selector
+        )]}}}}
+    ]
+    all(res)
+}
+
+test_bad_replicaset_under_deployment_minAvailable_int {
+    namespace := "default"
+    selector := {"matchLabels": {"key": "val"}}
+    labels := {"key": "val"}
+
+    # The replicaset and PDB does not allow for disruption, but this should not violate since replicaset is under control of a deployment
+    res = [test | test = count(k8srestrictpoddisruptionbudgets.violation) == 0
+        with input as input_wrap(generate_replicaset_under_deployment(
+            3,
+            namespace,
+            labels
+        ))
+        with data.inventory as {"namespace": {namespace: {"policy/v1": {"PodDisruptionBudget": [generate_pdb(
+            "minAvailable",
+            3,
+            namespace,
+            selector
+        )]}}}}
+    ]
+    all(res)
+}
+
+# --- Test different selectors ---
+
+test_matching_match_labels {
+	namespace := "default"
+    selector := {"matchLabels": {"key": "val"}}
+    labels := {"key": "val"}
+
+    count(k8srestrictpoddisruptionbudgets.violation) == 1
+        with input as input_wrap(generate_pod_controller(
+            "Deployment",
+			3,
```

Comment:
Indentation is a bit off here, right?

Same with the replica integer for most tests below

Reply:
Thanks, the issues was that I had tabs instead of spaces here.

For some reason my vscode sometimes wants to use tabs instead of spaces for the rego files. I have tried to fix it, but not quite managed yet.